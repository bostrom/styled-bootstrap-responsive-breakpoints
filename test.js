import React from "react";
import styled from "styled-components";
import renderer from "react-test-renderer";
import { matcher, serializer } from "jest-styled-components";

import { mediaBreakpointUpSm } from "./index";

expect.addSnapshotSerializer(serializer);
expect.extend(matcher);

test("It should match the mediaBreakpointUpSm snapshot", () => {
  const Wrapper = styled.section`
    padding: 4em;
    background: papayawhip;
  `;

  const Title = styled.h1`
    ${mediaBreakpointUpSm`
      color: blue;
    `}
  `;

  const tree = renderer.create(
    <Wrapper>
      <Title>Hello World, this is my worst styled component!</Title>
    </Wrapper>,
  ).toJSON();

  expect(tree).toMatchStyledComponentsSnapshot();
});

test("It should handle inline style objects snapshot", () => {
  const color = "red";
  const fontFamily = "'Comic Sans'";
  const Wrapper = styled.section`
    padding: 4em;
    background: papayawhip;
  `;

  const Title = styled.h1`
    ${mediaBreakpointUpSm`
      color: ${color};

      &:hover {
        font-family: ${fontFamily};
      }
    `}
  `;

  const tree = renderer.create(
    <Wrapper>
      <Title>Hello World, this is my worst styled component!</Title>
    </Wrapper>,
  ).toJSON();

  expect(tree).toMatchStyledComponentsSnapshot();
});
