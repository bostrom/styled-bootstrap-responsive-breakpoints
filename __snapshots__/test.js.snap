// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should handle inline style objects snapshot 1`] = `
.kGODAA {
  padding: 4em;
  background: papayawhip;
}

@media (min-width: 576px) {
  .eMvLgl {
    color: red;
  }

  .eMvLgl:hover {
    font-family: 'Comic Sans';
  }
}

<section
  className="kGODAA"
>
  <h1
    className="eMvLgl"
  >
    Hello World, this is my worst styled component!
  </h1>
</section>
`;

exports[`It should match the mediaBreakpointUpSm snapshot 1`] = `
.kGODAA {
  padding: 4em;
  background: papayawhip;
}

@media (min-width: 576px) {
  .kSTsOk {
    color: blue;
  }
}

<section
  className="kGODAA"
>
  <h1
    className="kSTsOk"
  >
    Hello World, this is my worst styled component!
  </h1>
</section>
`;
